api/lib/application/security-pre-handlers.js

```diff
@@ -300,6 +301,30 @@ async function checkUserIsAdminInSUPOrganizationManagingStudents(request, h) {
   return _replyForbiddenError(h);
 }
 
+async function checkUserBelongsToLearnersOrganization(request, h) {
+  if (!request.auth.credentials || !request.auth.credentials.userId) {
+    return _replyForbiddenError(h);
+  }
+
+  const userId = request.auth.credentials.userId;
+  const organizationLearnerId = parseInt(request.params.id);
+
+  let belongsToLearnersOrganization;
+
+  try {
+    belongsToLearnersOrganization = await checkUserBelongsToLearnersOrganizationUseCase.execute(
+      userId,
+      organizationLearnerId
+    );
+  } catch (e) {
+    return _replyForbiddenError(h);
+  }
+  if (belongsToLearnersOrganization) {
+    return h.response(true);
+  }
+  return _replyForbiddenError(h);
```

Comment:
```suggestion
  try {
    const belongsToLearnersOrganization = await checkUserBelongsToLearnersOrganizationUseCase.execute(
      userId,
      organizationLearnerId
    );
    if (belongsToLearnersOrganization) {
      return h.response(true);
    }
  } finally {
    return _replyForbiddenError(h);
  }
```

Reply:
Le refacto ne fonctionne pas. J'avais un doute de ce que je lisais [ici](https://levelup.gitconnected.com/5-things-you-dont-know-about-try-catch-finally-in-javascript-5d661996d77c). Et j'ai eu confirmation [là](https://eslint.org/docs/latest/rules/no-unsafe-finally). Le return de la response ne sera jamais appelé... J'ai eu l'alerte eslint en changeant le code.

Reply:
Ouah merci beaucoup, je ne savais pas du tout le point sur le fonctionnement du finally avec le return 👍